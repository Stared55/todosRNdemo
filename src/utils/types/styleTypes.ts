import {
  FlexStyle,
  ShadowStyleIOS,
  TextStyleAndroid,
  TextStyleIOS,
  TransformsStyle,
  TextStyle,
  ViewStyle,
} from 'react-native';

export type ViewStyles = ViewStyle &
  FlexStyle &
  ShadowStyleIOS &
  TransformsStyle;

export type TextStyles = TextStyle &
  TextStyleIOS &
  TextStyleAndroid &
  ViewStyle;
