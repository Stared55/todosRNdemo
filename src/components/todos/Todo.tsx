import * as React from 'react';
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import { Colors } from 'src/utils/styles/colors';
import { xScale, yScale } from 'src/utils/styles/scale';
import { ITodo } from 'src/utils/types/todoTypes';

export interface TodoProps {
  todo: ITodo;
  setSelectedIds: (id: number) => void;
}

const Todo: React.SFC<TodoProps> = ({ todo, setSelectedIds }) => {
  const handleOnPress = () => {};
  const toggleActiveStatus = () => {
    todo.selected = !todo.selected;
    setSelectedIds(todo.id);
  };

  return (
    <TouchableOpacity
      style={{
        ...styles.todoWrapper,
        shadowColor: todo.selected ? Colors.blue : 'rgba(0,0,0,0.4)',
      }}
      onPress={handleOnPress}
      onLongPress={toggleActiveStatus}>
      <View style={styles.leftColumn}>
        <Text>{todo.id}</Text>
      </View>
      <View style={styles.centerColumn}>
        <Text>{todo.title}</Text>
      </View>
      <View style={styles.rightColumn}>
        <Text>{todo.completed ? 'done' : 'to do'}</Text>
      </View>
    </TouchableOpacity>
  );
};

export default Todo;

const styles = StyleSheet.create({
  todoWrapper: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    width: '90%',
    marginVertical: xScale(6),
    height: yScale(40),
    backgroundColor: Colors.white,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.6,
    shadowRadius: 3.84,
    elevation: 5,
  },
  leftColumn: {
    width: '10%',
    justifyContent: 'center',
    alignItems: 'center',
    height: '100%',
  },
  centerColumn: {
    paddingLeft: 6,
    width: '75%',
    flexDirection: 'row',
    justifyContent: 'flex-start',
    height: '100%',
    alignItems: 'center',
  },
  rightColumn: {
    width: '15%',
    justifyContent: 'center',
    height: '100%',
    alignItems: 'center',
  },
});
