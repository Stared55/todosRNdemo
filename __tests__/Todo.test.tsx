import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Todo from 'src/components/todos/Todo';
import { ITodo } from 'src/utils/types/todoTypes';

describe('Todo', () => {
  const todo: ITodo = {
    id: 1,
    userId: 1,
    title: 'lorem',
    completed: false,
  };
  const handler = jest.fn();
  let component: ShallowWrapper;
  let props: any;

  beforeEach(() => {
    component = shallow(<Todo todo={todo} setSelectedIds={handler} />);
    props = component.props();
    jest.clearAllMocks();
  });

  it('should match to snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('should display correct todo title', () => {
    const labelValue = component.find({ children: 'lorem' });
    expect(labelValue).toHaveLength(1);
  });

  it('should display correct todo id', () => {
    const id = component.find({ children: 1 });
    expect(id).toHaveLength(1);
  });

  it('should display correct todo userId', () => {
    const id = component.find({ children: 1 });
    expect(id).toHaveLength(1);
  });

  it('should display correct todo completed state', () => {
    const id = component.find({ children: 'to do' });
    expect(id).toHaveLength(1);
  });

  it('should run callback', () => {
    component.simulate('longPress');
    expect(handler).toHaveBeenCalledTimes(1);
  });
});
