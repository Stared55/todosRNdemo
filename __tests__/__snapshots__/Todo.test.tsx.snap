// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo should match to snapshot 1`] = `
<ForwardRef
  onLongPress={[Function]}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "elevation": 5,
      "flexDirection": "row",
      "height": 78.47058823529412,
      "justifyContent": "center",
      "marginVertical": 12.857142857142858,
      "shadowColor": "rgba(0,0,0,0.4)",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.6,
      "shadowRadius": 3.84,
      "width": "90%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "center",
        "width": "10%",
      }
    }
  >
    <Text>
      1
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": "100%",
        "justifyContent": "flex-start",
        "paddingLeft": 6,
        "width": "75%",
      }
    }
  >
    <Text>
      lorem
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "center",
        "width": "15%",
      }
    }
  >
    <Text>
      to do
    </Text>
  </View>
</ForwardRef>
`;
